appinventor/components-ios/src/MagneticFieldSensor.swift

```diff
@@ -0,0 +1,92 @@
+// -*- mode: swift; swift-mode:basic-offset: 2; -*-
+// Copyright 2023 MIT, All rights reserved
+// Released under the Apache License, Version 2.0
+// http://www.apache.org/licenses/LICENSE-2.0
+
+import Foundation
+import CoreMotion
+
+class MagneticFieldSensor: NonvisibleComponent {
+  private let motionManager = CMMotionManager()
+  private var enabled : Bool = true
+  
+  @objc var AbsoluteStrength: Double {
+    guard let magnetometerData = motionManager.magnetometerData?.magneticField else {
+      return 0.0
+    }
+    let xStrength = Float(magnetometerData.x)
+    let yStrength = Float(magnetometerData.y)
+    let zStrength = Float(magnetometerData.z)
+    return sqrt(Double(xStrength * xStrength + yStrength * yStrength + zStrength * zStrength))
+  }
+  @objc var Available: Bool {
+    return motionManager.isMagnetometerAvailable
+  }
+  
+  @objc var XStrength: Float {
+    return Float(motionManager.magnetometerData?.magneticField.x ?? 0.0)
+  }
+  
+  @objc var YStrength: Float {
+    return Float(motionManager.magnetometerData?.magneticField.y ?? 0.0)
+  }
+  
+  @objc var ZStrength: Float {
+    return Float(motionManager.magnetometerData?.magneticField.z ?? 0.0)
+  }
+  
+  override init(_ container: ComponentContainer) {
+    super.init(container)
+    startListening()
```

Comment:
Rather than starting the listener here, implement `@objc open func Initialize()` so that the listening can start during the initialization phase, and only if `enabled`. Otherwise, this may prompt the user for permission even if Enabled = false is set by the app developer at the start of the app.